src/providers/Auth/Auth.provider.jsx

```diff
@@ -1,42 +1,45 @@
-import React, { useState, useEffect, useContext, useCallback } from 'react';
+import React, { useEffect, useContext, useState, useCallback } from 'react';
+import { USERS } from '../../constants';
 
-import { AUTH_STORAGE_KEY } from '../../utils/constants';
-import { storage } from '../../utils/storage';
-
-const AuthContext = React.createContext(null);
+const UserContext = React.createContext(null);
 
 function useAuth() {
-  const context = useContext(AuthContext);
+  const context = useContext(UserContext);
   if (!context) {
-    throw new Error(`Can't use "useAuth" without an AuthProvider!`);
+    throw new Error(`Can't use "useAuth" without an AuthContext!`);
   }
   return context;
 }
 
 function AuthProvider({ children }) {
-  const [authenticated, setAuthenticated] = useState(false);
-
-  useEffect(() => {
-    const lastAuthState = storage.get(AUTH_STORAGE_KEY);
-    const isAuthenticated = Boolean(lastAuthState);
-
-    setAuthenticated(isAuthenticated);
+  const [user, setUser] = useState(null);
+
+  const login = useCallback((name, password) => {
+    if (name in USERS && password === USERS[name].password) {
+      const { u } = USERS[name];
+      sessionStorage.setItem('session', JSON.stringify(u));
```

Comment:
I like that you implemented sessionStorage for login, I suggest on the next deliverable you could add as comment extras that you implemented 😉